libflux/flux-core/src/formatter/mod.rs

```diff
@@ -226,22 +225,25 @@ impl Formatter {
             self.write_rune(sep);
             self.write_rune(sep);
         }
-
         let mut prev: i8 = -1;
+        let mut previous_location: i32 = -1;
         for (i, stmt) in (&n.body).iter().enumerate() {
             let cur = stmt.typ();
             if i != 0 {
+                let current_location: i32 = stmt.base().location.start.line as i32;
+                //compare the line position of adjacent lines to preserve formatted double new lines
+                let line_gap = current_location - previous_location;
                 self.write_rune(sep);
                 // separate different statements with double newline or statements with comments
-                if cur != prev || starts_with_comment(Node::from_stmt(&stmt)) {
+                if line_gap > 1 || cur != prev || starts_with_comment(Node::from_stmt(&stmt)) {
                     self.write_rune(sep);
                 }
```

Comment:
This is the kind of thing that reasonable people may disagree on, i.e. a [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding) kind of issue. Personally I like the logic here but I would make it so we don't add a blank line between different kinds of statements. E.g., this looks fine to me:
```js
fun = (a, b) => {
  c = a * b
  return c
}
```
I don't think it helps readability to put a blank line between the two statements in the function body.

This is a different change from the issue you're fixing here, so no need to fix it now.

Reply:
I agree, the `cur != prev` should come out after this change is in. So long as gaps are preserved where they were originally, there is no need to insert whitespace if the statement types differ.